gci/issues.py

```diff
@@ -0,0 +1,580 @@
+    'research',
+]
+
+
+def issue_is_available(issue, issue_metadata):
+    logger = logging.getLogger(__name__ + '.issue_is_available')
+
+    issue_id = issue.number
+    url = issue.web_url
+
+    logger.debug(f'checking {issue_id} availability...')
+
+    if str(issue.state) in ('closed', ):
+        logger.info(f'{issue_id}: {issue.state}')
+        return False
+
+    if str(issue.state) not in ('open', ):
+        logger.warning(f'{url}: Unexpected state {issue.state}')
+        return False
+
+    if issue.assignees:
+        assignees = [assignee.username for assignee in issue.assignees]
+        if len(assignees):
+            assignees = assignees[0]
+        logger.warning(f'Skipping {url} assigned to {assignees}')
+        return False
+
+    assert issue.title
+
+    label_names = set(issue.labels)
+
+    required_label = issue_metadata.get('required_label')
+    if required_label and required_label not in label_names:
+        logger.warning(
+            f'Skipping {issue_id} due to missing label {required_label}')
+        return False
+
+    if label_names:
+        wrong_labels = label_names.intersection(unsuitable_labels)
+        if wrong_labels:
+            logger.warning(
+                f'Skipping {issue_id} due to labels {",".join(wrong_labels)}')
+            return False
+
+    logger.info(f'{issue_id} is available')
+
+    return True
+
+
+def generate_task(issue, metadata):
+    logger = logging.getLogger(__name__ + '.generate_task')
+
+    title = issue.title
+    title = title.replace('"', '').replace('!', '').replace('  ', ' ').strip()
+
+    logger.info(f'generating task for issue {issue.number}: {title}')
+    url = issue.web_url
+
+    logger.debug(f'{url}: {title}')
+
+    repo_name = metadata['repo_name']
+    short_name = metadata['short_name']
+    mentors = metadata.get('mentors')
+    difficulty_level = metadata['difficulty_level']
+
+    tags = set(metadata.get('tags', []))
+
+    tags.add('issues')
+
+    if 'rst' in tags:
+        tags.add('reStructuredText')
+        tags.remove('rst')
+    if 'restructuredtext' in tags:
+        tags.add('reStructuredText')
+        tags.remove('restructuredtext')
+
+    label_names = issue.labels
+
+    logger.info(f'labels {",".join(label_names)}')
+
+    if 'documentation' in label_names or 'area/documentation' in label_names:
+        tags.add('documentation')
+    if 'area/tests' in label_names:
+        tags |= {'pytest', 'unittest', 'unit test', 'unit tests'}
+
+    if 'hoster/gitlab' in label_names:
+        tags |= {'gitlab'}
+    elif 'hoster/github' in label_names:
+        tags |= {'github'}
+    if '.coafile' in title:
+        tags |= {'.coafile'}
+
+    if difficulty_level == 'newcomer':
+        if metadata['host'] == 'gitlab.com':
+            tags.add('GitLab')
+        else:
+            tags.add('git')
+
+    if 'README.md' in title:
+        tags.add('markdown')
+        assert difficulty_level == 'newcomer'
+    elif '.md' in title:
```

Comment:
`'docs'` could be a keyword to look for.

Reply:
Yup.  Nice.